### ISSUE TITLE
How to set the dataset in CDD, to get some suitable Precision

### ISSUE THREAD

**Issue by eedavidwu** [lehaifeng/DASNet#10]
Hi, I run this work with precision less that 8% in 10 epoch (also it includes some TN results, so this change map is useless....). 
And I dont know why for that, and could u please tell me how did u set the dataset.
In my experiments, I used the pretrained model to train the CDD real subset data directly， not used model data (synthesized data). Did u use all of the CDD dataset including model and real?

**Comment by SherryZhaoXY** [comment]
> Hi, I run this work with precision less that 8% in 10 epoch (also it includes some TN results, so this change map is useless....).
> And I dont know why for that, and could u please tell me how did u set the dataset.
> 
> Did u use all of the CDD dataset including model and real...
> 
> And the interesting results show that the changemap from the conv5 is much better than the embading (which means Spatial attention Mechanism makes sense but the channel attention does not)...Did u face the same situation...

Hello, sir, it seems like you have got a result, could you please share your test.py to me?  I wrote one myself, but it doesn't work. Thank you so much!

**Comment by eedavidwu** [comment]
> > Hi, I run this work with precision less that 8% in 10 epoch (also it includes some TN results, so this change map is useless....).
> > And I dont know why for that, and could u please tell me how did u set the dataset.
> > Did u use all of the CDD dataset including model and real...
> > And the interesting results show that the changemap from the conv5 is much better than the embading (which means Spatial attention Mechanism makes sense but the channel attention does not)...Did u face the same situation...
> 
> Hello, sir, it seems like you have got a result, could you please share your test.py to me? I wrote one myself, but it doesn't work. Thank you so much!

I didnt write a test.py and directly use the function: single_layer_similar_heatmap_visual (where u can check the code).
(Make sure that u train correctly such that the loss and some other issues disappear). 
U can make some changes to suit ur dataset format..(For that in this way, everyone's situation maybe different.)

**Comment by SherryZhaoXY** [comment]
Oh I got you.Thank you for your reply!

**Comment by lehaifeng** [comment]
> Hi, I run this work with precision less that 8% in 10 epoch (also it includes some TN results, so this change map is useless....).
> And I dont know why for that, and could u please tell me how did u set the dataset.
> 
> Did u use all of the CDD dataset including model and real...
> 
> And the interesting results show that the changemap from the conv5 is much better than the embading (which means Spatial attention Mechanism makes sense but the channel attention does not)...Did u face the same situation...

Hi，I only used the remote sensing data in the CDD dataset and you can use validate function to get the result.In my settings, every 1000 batches will be tested.

**Comment by eedavidwu** [comment]
> Hi，I only used the remote sensing data in the CDD dataset and you can use validate function to get the result.In my settings, every 1000 batches will be tested.

Thanks a lot-. (Maybe the precision problems come from my rewritten or the validate functions- I would check it again.)

**Comment by eedavidwu** [comment]
> > Hi, I run this work with precision less that 8% in 10 epoch (also it includes some TN results, so this change map is useless....).
> > And I dont know why for that, and could u please tell me how did u set the dataset.
> > Did u use all of the CDD dataset including model and real...
> > And the interesting results show that the changemap from the conv5 is much better than the embading (which means Spatial attention Mechanism makes sense but the channel attention does not)...Did u face the same situation...
> 
> Hi，I only used the remote sensing data in the CDD dataset and you can use validate function to get the result.In my settings, every 1000 batches will be tested.

Hi, I want to check how did you compute the precise and recall. 
For that my results shows the algorithm works as the paper shows, but the precise maintain around 60% in Val dataset. (I compute TN,TP,FN,FP pixel by pixel)...Is the precise computated by the same way in this paper? (For that I check the code, it computed the distance first, and then use eval function to compute true or false each pixel or one IOU?)
